core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -437,6 +438,37 @@ private[spark] class SparkSubmit extends Logging {
       }.orNull
     }
 
+    def resolveMainClassIfNeeded(jarPath: String): Try[String] = {
```

Comment:
I think this looks better inlined below in the only place where it's called; also using a normal `try...catch` instead of a `Try` + match.